pipeline/extract.py

```diff
@@ -72,56 +75,72 @@ async def extract_list_of_items(event_list: list[dict],
             for item in event['items']:
                 if not (item["item_type"] == "a" or item["item_type"] == "t"):
                     continue
-                tasks.append(extract_and_scrape_item(session, item, timeout))
+                tasks.append(extract_and_scrape_item(
+                    semaphore, session, item, timeout))
 
         item_list = await asyncio.gather(*tasks)
 
     return item_list
 
 
-async def extract_and_scrape_item(session: aiohttp.ClientSession,
+async def extract_and_scrape_item(semaphore: asyncio.Semaphore,
+                                  session: aiohttp.ClientSession,
                                   purchase_dict: dict,
-                                  timeout: int) -> dict:
+                                  timeout: int,
+                                  delay: float = DELAY_BETWEEN_REQUESTS) -> dict:
     '''Scrape and extract information for an item, giving
     more complete information for the item'''
-    if purchase_dict["item_type"] == "t":
-        purchase_dict["track_tags"] = await scrape_tags(session,
-                                                        insert_protocol_url(
-                                                            purchase_dict["url"]),
-                                                        timeout)
-    if purchase_dict["item_type"] == "a":
-        purchase_dict["album_tags"] = await scrape_tags(session,
-                                                        insert_protocol_url(
-                                                            purchase_dict["url"]),
-                                                        timeout)
-    if purchase_dict["item_type"] == "t" and purchase_dict["album_title"] is not None:
-        stem_url = get_stem_url(insert_protocol_url(purchase_dict["url"]))
-        album_url = await scrape_album_url(session,
-                                           insert_protocol_url(
-                                               purchase_dict["url"]),
-                                           timeout)
-
-        if stem_url is not None and album_url is not None:
-            purchase_dict["album_url"] = stem_url + album_url
-            purchase_dict["album_tags"] = await scrape_tags(session, stem_url + album_url, timeout)
```

Comment:
Why are album_tags not being scraped for songs belonging to albums?

Reply:
It can often give a 429 error when you scrape too many albums/tracks. The albums also tend to have the exact same tags as the tracks, making them redundant in many cases.

Reply:
Which is why it was removed in later commits